### ISSUE TITLE
Issue with  jit_runtimeInstrumentation & port_runtimeInstrumentation

### ISSUE THREAD

**Issue by ChengJin01** [eclipse/openj9#1018]
Given that and all the low level support on RI was removed, RI isn't going to work for openj9.
In such case, it should be feasible to turn off jit_runtimeInstrumentation and port_runtimeInstrumentation in specs on zOS.

However, the compilation result indicates it fails with the following error in JIT after turning off these two flags:
CCN0793(I) Compilation failed for file openj9/runtime/compiler/../compiler/trj9/z/codegen/J9TreeEvaluator.cpp.  Object file not created.
openj9/runtime/compiler/../compiler/trj9/build/rules/zos-xlc/common.mk:99: recipe for target 'openj9/runtime/compiler/../objs/compiler/trj9/z/codegen/J9TreeEvaluator.o' failed
make[3]: *** [openj9/runtime/compiler/../objs/compiler/trj9/z/codegen/J9TreeEvaluator.o] Error 12
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory 'openj9/runtime/compiler/trj9'
Makefile:83: recipe for target 'trj9' failed
make[2]: *** [trj9] Error 2
make[2]: Leaving directory 'openj9/runtime/compiler/trj9'
makefile:57: recipe for target 'default' failed
make[1]: *** [default] Error 2
make[1]: Leaving directory 'openj9/runtime/compiler'
makefile:1138: recipe for target 'j9jitlauncher' failed
make: *** [j9jitlauncher] Error 2

We are not sure what's going to happen at runtime if the JIT tries to use it

**Comment by ChengJin01** [comment]
FYI:  @pshipton,  @DanHeidinga

**Comment by dsouzai** [comment]
I'm guessing it's failing because of code in this method, namely the `J9_JIT_TOGGLE_RI_IN_COMPILED_CODE` flag:
https://github.com/eclipse/openj9/blob/26f8c7edb1fe37253d001243c0ce14473334d07f/runtime/compiler/trj9/z/codegen/J9TreeEvaluator.cpp#L9876-L9927

Calls to `generateRuntimeInstrumentationOnOffSequence` seem to be guarded by `cg()->getSupportsRuntimeInstrumentation()`, eg:

https://github.com/eclipse/openj9/blob/26f8c7edb1fe37253d001243c0ce14473334d07f/runtime/compiler/trj9/z/codegen/J9S390PrivateLinkage.cpp#L1500-L1501

However, `setSupportsRuntimeInstrumentation` is only called when `TR_HWProfilerDisableRIOverPrivateLinkage` is enabled via the -Xjit command line:
https://github.com/eclipse/openj9/blob/26f8c7edb1fe37253d001243c0ce14473334d07f/runtime/compiler/trj9/z/codegen/J9CodeGenerator.cpp#L182-L190

Other code that emits `RION`/`RIOFF` is this:
https://github.com/eclipse/openj9/blob/26f8c7edb1fe37253d001243c0ce14473334d07f/runtime/compiler/trj9/z/codegen/J9ZSnippet.cpp#L28-L54
which is also guarded by `getSupportsRuntimeInstrumentation`.

In other words, none of this code has been active for a very long time. @joransiu fyi.
